### ISSUE TITLE
Azkaban 3.38.0 removing executor from databse during shutdown. For this version we have enabled multi executor mode . Also inserted value for executor with active flag in database. However when we shutdown executor it removes value of executor from database and on starting creates it with new key. In such scenario our scheduled flows are failing since id key for executor is updated.

### ISSUE THREAD

**Issue by monanik39** [azkaban/azkaban#1581]

**Comment by monanik39** [comment]
PFA screenshot before restart
![before-restart](https://user-images.githubusercontent.com/27038620/33804223-5edd3726-dddb-11e7-8240-1b2219a5f182.PNG)

After shutdown

![executor-shutdown](https://user-images.githubusercontent.com/27038620/33804225-68684880-dddb-11e7-83c8-04a8538963d7.PNG)

after restarting
![after-exec-restart](https://user-images.githubusercontent.com/27038620/33804227-6d8258c4-dddb-11e7-9f82-e065310276fe.PNG)

**Comment by leoChaoGlut** [comment]
You can activate an executor by hand: 
`curl http://${executorHost}:${executorPort}/executor?action=activate`

**Comment by monanik39** [comment]
Yup but after every restart why is executor ID changing? In previous versions this behavior was not observed. Is any additional setting required to override this behavior. 
For my setup once executors are inserted i do not wish to have their id changed due to restart.

**Comment by monanik39** [comment]
is this feature of azkaban if you are using azkaban in multi executor mode?

**Comment by kunkun-tang** [comment]
>  our scheduled flows are failing since id key for executor is updated.
Do your scheduled flows specify a specific executor id? and why?

@monanik39

**Comment by monanik39** [comment]
Yes currently we specify executor id in flow. Cause we want to use same executor for running job of 2 projects. So we have static executor mapping as per project.

**Comment by monanik39** [comment]
Also recreated executors are inactive by default(active =0 in database). is there any property which will set active executor flag to 1  once it is recreated? Cause as per my understanding new entry will be created only if server is up so active flag can be 1 by default?

**Comment by FrommyMind** [comment]
@monanik39 
 Also recreated executors are inactive by default(active =0 in database). is there any property which will set active executor flag to 1 once it is recreated? Cause as per my understanding new entry will be created only if server is up so active flag can be 1 by default?

Have you solved this question?

**Comment by FrommyMind** [comment]
Can we just set the default value of the  column "active" in "executor" table  to 1?
